src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -969,7 +969,7 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
             var fullName = GetFullHtmlFieldName(viewContext, expression);
             if (string.IsNullOrEmpty(fullName))
             {
-                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(expression));
+                throw new ArgumentException(Resources.HtmlGenerator_FieldNameCannotBeNullOrEmpty);
```

Comment:
This is the only change you're actually testing. Update `DefaultHtmlGeneratorTest` (which tests `GetCurrentValues()`), `HtmlHelperCheckboxTest`, `HtmlHelperSelectTest`, the relevant tag helper tests e.g. `TextAreaTagHelperTest`, and perhaps `DefaultEditorTemplatesTest` to increase coverage. (Is :ok: that the HTML helper tests don't exercise `TextArea` and `ValidationMessage` since the corresponding `IHtmlGenerator` methods will be tested through the tag helpers.)